https://github.com/ziatdinovmax/pyroVED/issues/54

utkarshp1161 opened the pull request:
Based on Discussion:
https://github.com/ziatdinovmax/pyroVED/discussions/53

Commit:
added VAE_gp notebook

Commit:
Revert "added VAE_gp notebook"

This reverts commit 3b8bf13ab4b80be4c705dbdb1ec3d774bce586bb.

Commit:
added VAE_gp notebook

utkarshp1161 committed:
Update VAE_gp.ipynb

Commit:
add: predict_on_latent

ziatdinovmax reviewed (changes requested):
A few comments/suggestions:
- Please add the unit tests to cover new code lines
- I suggest moving GP to a separate utility function and importing it from there. Then you just call something like gpr(*args, **kwargs).
- What is the meaning of training_data? Is it the same as the one used to train VAE? Or can it be any X? Need to explain it clearly.
- Making predictions on a z grid without any connection to input space may not be very useful. Let's add a decoder part that will show how these z-points look in the original data space.

ziatdinovmax commented:
@utkarshp1161 Any progress on this one?

utkarshp1161 commented:
Hi Max,

Was busy with qualifiers this past week. Will get on this.

Commit:
add: gp

utkarshp1161 commented:
Hi Max,

1. https://github.com/utkarshp1161/pyroVED/blob/main/pyroved/utils/gp.py has the gp model
2. https://github.com/utkarshp1161/pyroVED/blob/main/pyroved/models/ivae.py has the incorporated  def predict_on_latent() funtion
3. Notebook used: https://github.com/utkarshp1161/pyroVED/blob/main/examples/VAE_gp.ipynb

wrt tests:

I am getting below error when I run:  pytest test/:
see error.txt file attached
[error.txt](https://github.com/ziatdinovmax/pyroVED/files/14078126/error.txt)

utkarshp1161 commented:
> Hi Max,
> 
> 1. https://github.com/utkarshp1161/pyroVED/blob/main/pyroved/utils/gp.py has the gp model
> 2. https://github.com/utkarshp1161/pyroVED/blob/main/pyroved/models/ivae.py has the incorporated  def predict_on_latent() funtion
> 3. Notebook used: https://github.com/utkarshp1161/pyroVED/blob/main/examples/VAE_gp.ipynb
> 
> wrt tests:
> 
> 
> I am getting below error when I run: pytest test/: see error.txt file attached [error.txt](https://github.com/ziatdinovmax/pyroVED/files/14078126/error.txt)
  EDIT: resolved: Environment was pointing to local python env rather than conda env. pytest running but below error
  [error2.txt](https://github.com/ziatdinovmax/pyroVED/files/14078319/error2.txt)

> 
> 
> I am thinking of adding tests similar to manifold2d test in [test_model.py](https://github.com/utkarshp1161/pyroVED/blob/main/tests/test_models.py). Also for testing gp part I am thinking of adding a test_utils.py in https://github.com/utkarshp1161/pyroVED/blob/main/tests/
> 
> Do you have suggestions on what to include in tests?

ziatdinovmax reviewed (changes requested):
Thanks for the update/revision.

Several comments:
- In ivae.py, on line 329, you seem to be overwriting the gp_iterations parameter and setting it to 1.
- In the utils/gp.py, the docstring says that you define and *train* a GP model, but there's no actual training.
- I suggest that the ```predict_on_latent``` method returns both the GP predictions and the manifold2d tensor, whole the plotting is optional.
- For tests, just add a separate test for the (modified) GP utility to confirm that it runs and returns a tensor with an expected shape. Then, do the same for the predict_on_latent.

Commit:
add: test, feature: ivae.predict_on_labels

utkarshp1161 commented:
> Thanks for the update/revision.
> 
> Several comments:
> 
> * In ivae.py, on line 329, you seem to be overwriting the gp_iterations parameter and setting it to 1.
Thanks for pointing out.
> * In the utils/gp.py, the docstring says that you define and _train_ a GP model, but there's no actual training.
Yep. CHanged
> * I suggest that the `predict_on_latent` method returns both the GP predictions and the manifold2d tensor, whole the plotting is optional.
Added a "plot" boolean which defaults to False. Returns  GP predictions and the manifold2d tensor
> * For tests, just add a separate test for the (modified) GP utility to confirm that it runs and returns a tensor with an expected shape. Then, do the same for the predict_on_latent.
Yep added.

ziatdinovmax commented on pyroved/models/ivae.py:
Almost there... Instead of returning z, we should be returning z_decoded, which is z passed through a trained decoder. In this particular case, we can obtain it simply as z_decoded = self.manifold2d(d, plot=False).

I'm curious if there's a specific reason you don't want to train GP inside utils/gp.py?

utkarshp1161 replied:
> Almost there... Instead of returning z, we should be returning z_decoded, which is z passed through a trained decoder. In this particular case, we can obtain it simply as z_decoded = self.manifold2d(d, plot=False).

I was having something else in mind, basically having the latent coordinates rather than decoded ones. Changed to this.

> I'm curious if there's a specific reason you don't want to train GP inside utils/gp.py?

Changed training to utils. Right, this way its more modular.

ziatdinovmax replied:
> I was having something else in mind, basically having the latent coordinates rather than decoded ones. Changed as per this suggestion.

In principle, we can have all three. Something like
 ```python3
return (z, z_decoded), predictions
```

Let me know if you would like to add this. Other than that I'm ready to merge it.

Commit:
gp trains in gp.py

ziatdinovmax commented on error2.txt:
This error.2txt file is unnecessary (should be removed)

utkarshp1161 replied:
My bad, removed.

Commit:
remove error.txt

Commit:
mod: (z, z_decoded) in ivae.predict_on_latent

Commit:
mod: (z, z_decoded) in ivae.predict_on_latent
